packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -1166,48 +1351,37 @@ class TextPainter {
   ///
   /// Valid only after [layout] has been called.
   double? getFullHeightForCaret(TextPosition position, Rect caretPrototype) {
-    _computeCaretMetrics(position, caretPrototype);
-    return _caretMetrics.fullHeight;
+    final _CaretMetrics caretMetrics = _computeCaretMetrics(position);
+    return caretMetrics is _LineCaretMetrics ? caretMetrics.fullHeight : null;
```

Comment:
Looks like this will return null if `caretMetrics` is `_EmptyLineCaretMetrics` which according to the comments also happens in a line between two newlines.
Is this desired? I think in this case we should return a valid caret height as we know the font size in this very `TextSpan`. Otherwise (e.g. a tiny font), the caller may assume completely wrong caret height.

Reply:
I believe I didn't change the current behavior. The current implementation isn't correct but it's probably for a different PR.

Reply:
Maybe add an TODO linking to an issue?